### PR TITLE
Improve CLI usage messages

### LINKED ISSUE #8962: CLI: Ambiguous usage help for positional parameters since #8878
#### Problem

Since https://github.com/solana-labs/solana/pull/8878, many CLI commands generate ambiguous usage help for positional parameters.  eg. https://docs.solana.com/v/master/cli/usage#solana-authorize-nonce-account

#### Proposed Solution

`<ROLE TYPE>` for positional parameters, flags self-document optional parameters

cc/ @garious @CriesofCarrots

### PR DESCRIPTION
**Pull request by garious**
#### Problem

I caused a usability regression in the CLI usage messages when I changed the value_names to types.

#### Summary of Changes

* Qualify the value name in positional arguments
* Tighten some help messages along the way

Fixes #8962

### CODE REVIEW

File: cli/src/stake.rs [hunk 1]
```diff
@@ -255,24 +255,24 @@ impl StakeSubCommands for App<'_, '_> {
         )
         .subcommand(
             SubCommand::with_name("split-stake")
-                .about("Split a stake account")
+                .about("Create a new stake account using tokens from an existing one")
```
Comment (CriesofCarrots) [inline cli/src/stake.rs]: Wonder if this should be more explicit. This sounds like it creates a new "clean" stake account, but in fact the the state of the original account (delegated, stake warmed up, lockup, etc) is duplicated.
I think the word "split" is better than "create" in that regard.

**Comment by t-nelson** [comment]
I pushed a couple stragglers

There are some contexts where `<NUMBER>` seems to fall short on its own.  eg. if you call `create-*-account` with the wrong number of positional args,
```
$ solana create-stake-account
error: The following required arguments were not provided:
    <ACCOUNT_KEYPAIR>
    <NUMBER>

USAGE:
    solana create-stake-account <ACCOUNT_KEYPAIR> <NUMBER> --config <FILEPATH>

For more information try --help
```

`<AMOUNT_NUMBER>` seems redundant. `<SOL_NUMBER>` doesn't necessarily clear things up w/o domain knowledge. `<AMOUNT>` still feels best here. wdyt?

File: cli/src/stake.rs [hunk 2]
```diff
@@ -255,29 +255,29 @@ impl StakeSubCommands for App<'_, '_> {
         )
         .subcommand(
             SubCommand::with_name("split-stake")
-                .about("Split a stake account")
+                .about("Duplicate a stake account, splitting the tokens between the two")
```
Comment (garious) [inline cli/src/stake.rs]: @CriesofCarrots, better?

Comment (CriesofCarrots) [reply cli/src/stake.rs]: Much!

**Pull request merged**